warcio/checker.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import print_function
+
+from warcio.archiveiterator import ArchiveIterator
+
+
+class Checker(object):
+    def __init__(self, cmd):
+        self.inputs = cmd.inputs
+        self.verbose = cmd.verbose
+        self.exit_value = 0
+
+    def process_all(self):
+        for filename in self.inputs:
+            self.process_one(filename)
+        return self.exit_value
+
+    def process_one(self, filename):
+        with open(filename, 'rb') as stream:
+            self.file_printed = False
```

Comment:
I think we should define all instance properties in `__init__`  with sane default values before usage

Reply:
I am following the style of indexer.py, for this and the next comment.

Reply:
Yeah, weshould init file_printed and filename in the __init__
If indexer.py is missing this, its a mistake, should fix too.

Reply:
actually self.fileprinted was just a helper function that was confusingly implemented, I've gotten rid of the instance properties related to it.